pkg/collector/corechecks/containers/containerd.go

```diff
@@ -336,12 +345,28 @@ func parseAndSubmitMem(metricName string, sender aggregator.Sender, stat *v1.Mem
 
 }
 
-func computeCPU(sender aggregator.Sender, cpu *v1.CPUStat, tags []string) {
+func computeCPU(sender aggregator.Sender, cpu *v1.CPUStat, cpuLimits *specs.LinuxCPU, startTime, currentTime time.Time, tags []string) {
+	if cpu == nil || cpu.Usage == nil {
+		return
+	}
+
 	sender.Rate("containerd.cpu.system", float64(cpu.Usage.Kernel), "", tags)
 	sender.Rate("containerd.cpu.total", float64(cpu.Usage.Total), "", tags)
 	sender.Rate("containerd.cpu.user", float64(cpu.Usage.User), "", tags)
-	sender.Rate("containerd.cpu.throttled.periods", float64(cpu.Throttling.ThrottledPeriods), "", tags)
 
+	if cpu.Throttling != nil {
+		sender.Rate("containerd.cpu.throttled.periods", float64(cpu.Throttling.ThrottledPeriods), "", tags)
+		sender.Rate("containerd.cpu.throttled.time", float64(cpu.Throttling.ThrottledTime), "", tags)
+	}
+
+	timeDiff := float64(currentTime.Sub(startTime).Nanoseconds()) // cpu.total is in nanoseconds
+	if timeDiff > 0 {
+		cpuLimitPct := float64(runtime.NumCPU())
+		if cpuLimits != nil && cpuLimits.Period != nil && cpuLimits.Quota != nil && *cpuLimits.Quota > 0 {
+			cpuLimitPct = float64(*cpuLimits.Quota) / float64(*cpuLimits.Period)
```

Comment:
should we check that `*cpuLimits.Period` > 0 ?

Reply:
It should be impossible to have `cpuLimits.Period == 0` with `cpuLimits.Quota > 0`, but in doubt I'll add a check on it.

Reply:
Fixed